File: Data Structures & Algorithms/climbing-stairs/submission-4.java
class Solution {
    public int climbStairs(int n) {
        int one = 0, two = 1;
        
        for(int i=1; i<= n; i++){
            int three = one + two;
            one = two;
            two = three;
        }

        return two;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-4.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 0: return 0
        if n == 1: return nums[0]

        def helper(num):
            n = len(num)
            one, two = num[n-1], 0
            for i in range(n-2, -1, -1):
                three = max(one, two + num[i])
                two = one
                one = three
            return one

        return max(helper(nums[1:]), helper(nums[:len(nums)-1]))

